tests/test_connection.py

```diff
@@ -48,6 +57,9 @@ def test_sharing_connections(self):
 
         connect('mongoenginetest', alias='testdb2')
         actual_connection = get_connection('testdb2')
+
+        # horrible, but since PyMongo3+, connection are created asynchronously
+        sleep(0.1)
```

Comment:
This will need changing as you can almost guarantee it will break as soon as it hits a CI environment and everything is that much slower.